src/Microsoft.ML.DataView/VBuffer.cs

```diff
@@ -301,32 +395,35 @@ internal VBufferEditor<T> GetEditor()
             bool keepOldOnResize = false,
             bool requireIndicesOnDense = false)
         {
-            Contracts.CheckParam(newLogicalLength >= 0, nameof(newLogicalLength));
-            Contracts.CheckParam(valuesCount == null || valuesCount.Value <= newLogicalLength, nameof(valuesCount));
+            Contracts.CheckParam(newLogicalLength >= 0, nameof(newLogicalLength), "Must be non-negative.");
+            Contracts.CheckParam(valuesCount == null || valuesCount.Value >= 0, nameof(valuesCount),
+                "If specified, must be non-negative.");
+            Contracts.CheckParam(valuesCount == null || valuesCount.Value <= newLogicalLength, nameof(valuesCount),
+                "If specified, must be no greater than " + nameof(newLogicalLength));
 
-            valuesCount = valuesCount ?? newLogicalLength;
+            int logicallValuesCount = valuesCount ?? newLogicalLength;
```

Comment:
type-o `logicall`

Reply:
Sorry, that's intentional. Mr. Spock and I are starting a mobile wireless service provider, called "Logicall." We're going to bring the big players down! This is just a bit of viral guerrilla marketing he and I are doing.

It's going great, by the way; we save a lot of time by avoiding lengthy strategy meetings. A quick mind meld, and we're totally on the same page. The only real trouble is he insisted we make our branded Bluetooth ear-headsets *also* work with pointed ears. You wouldn't believe how hard that was to get right. We eventually went with a modular silicone "ear grip" that can be swapped out as needed. What a pain!

...I'm sorry, I'm very confused. I just had a very vivid dream last night and only just now realized it was not real. Correcting.

Reply:
^ this should be included in this  code comment., for future generations.

---
In reply to: [270549701](https://github.com/dotnet/machinelearning/pull/3136#discussion_r270549701) [](ancestors = 270549701)